packages/js/product-editor/src/components/prepublish-panel/schedule-section/schedule-section.tsx

```diff
@@ -19,102 +13,22 @@ import {
-	isSameDay,
-	isSiteSettingsTime12HourFormatted,
-	isSiteSettingsTimezoneSameAsDateTimezone,
-} from '../../../utils';
-
-export function getFormattedDateTime( value: string ) {
-	const { formats } = getSettings() as DateSettings;
-
-	return dateI18n(
-		sprintf(
-			// translators: %s: Time of day the product is scheduled for.
-			_x(
-				'F j, Y %s',
-				'product schedule full date format',
-				'woocommerce'
-			),
-			formats.time
-		),
-		value,
-		undefined
-	);
-}
-
-export function getFullScheduleLabel( dateAttribute: string ) {
-	const timezoneAbbreviation = getSiteSettingsTimezoneAbbreviation();
-	const formattedDate = getFormattedDateTime( dateAttribute );
-
-	return isRTL()
-		? `${ timezoneAbbreviation } ${ formattedDate }`
-		: `${ formattedDate } ${ timezoneAbbreviation }`;
-}
-
-export function getScheduleLabel( dateAttribute?: string, now = new Date() ) {
-	if ( ! dateAttribute ) {
-		return __( 'Immediately', 'woocommerce' );
-	}
-
-	// If the user timezone does not equal the site timezone then using words
-	// like 'tomorrow' is confusing, so show the full date.
-	if ( ! isSiteSettingsTimezoneSameAsDateTimezone( now ) ) {
-		return getFullScheduleLabel( dateAttribute );
-	}
-
-	const { formats } = getSettings() as DateSettings;
-	const date = getDate( dateAttribute );
-
-	if ( isSameDay( date, now ) ) {
-		return sprintf(
-			// translators: %s: Time of day the product is scheduled for.
-			__( 'Today at %s', 'woocommerce' ),
-			dateI18n( formats.time, dateAttribute, undefined )
-		);
-	}
-
-	const tomorrow = new Date( now );
-	tomorrow.setDate( tomorrow.getDate() + 1 );
-
-	if ( isSameDay( date, tomorrow ) ) {
-		return sprintf(
-			// translators: %s: Time of day the product is scheduled for.
-			__( 'Tomorrow at %s', 'woocommerce' ),
-			dateI18n( formats.time, dateAttribute, undefined )
-		);
-	}
-
-	if ( date.getFullYear() === now.getFullYear() ) {
-		return dateI18n(
-			sprintf(
-				// translators: %s: Time of day the product is scheduled for.
-				_x(
-					'F j %s',
-					'product schedule date format without year',
-					'woocommerce'
-				),
-				formats.time
-			),
-			date,
-			undefined
-		);
-	}
-
-	return getFormattedDateTime( dateAttribute );
-}
 
 export function ScheduleSection( { postType }: ScheduleSectionProps ) {
-	const [ editedDate, setDate, date ] = useEntityProp< string >(
-		'postType',
-		postType,
-		'date_created'
-	);
+	const [ productId ] = useProductEntityProp< number >( 'id' );
+	const { schedule, date, formattedDate } = useProductScheduled( postType );
+
+	// @ts-expect-error There are no types for this.
+	const { editEntityRecord } = useDispatch( 'core' );
 
-	function handlePublishDateTimePickerChange( value: string ) {
-		setDate( value );
+	async function handlePublishDateTimePickerChange( value: string | null ) {
+		await schedule( ( product ) => {
+			return editEntityRecord( 'postType', postType, productId, product );
+		}, value ?? undefined );
```

Comment:
Why not sending directly `null` instead `undefined` here?
It seems like we want that value to be `null` when https://github.com/woocommerce/woocommerce/pull/44783/files#diff-cd4cd07833719e5ac5d4aff2a6f2ac32a48e4f6ee84c93b2f789eed9922d1a61R38.

Reply:
because null is not a valid value.